crates/ruff_python_formatter/src/trivia.rs

```diff
@@ -39,3 +39,78 @@ pub(crate) fn find_first_non_trivia_character_in_range(
 
     None
 }
+
+/// Returns the number of newlines between `offset` and the first non whitespace character in the source code.
+#[allow(unused)] // TODO(micha) Remove after using for statements.
+pub(crate) fn lines_before(code: &str, offset: TextSize) -> u32 {
+    let head = &code[TextRange::up_to(offset)];
+    let mut newlines = 0u32;
+
+    for (index, c) in head.char_indices().rev() {
```

Comment:
what does the `.rev()` do here?

Reply:
We iterate the characters from the back rather than the front